src/states/adult/mod.rs

```diff
@@ -154,6 +154,12 @@ impl Adult {
         conn_infos: Vec<ConnectionInfo>,
         details: SignedRelocateDetails,
     ) -> Result<State, RoutingError> {
+        let their_key_infos = self
```

Comment:
This does not seem right, Adults don't have their_key_info populated, it needs to come from SignedRelocateDetails.
Am I missing something?

Reply:
I mean it only has our last Elders key infos, not any other section.

Reply:
Doesn't it come from `GenesisUpdate`?

Reply:
I believe the shared state is empty for now, but also it would be out of date.

Reply:
I don't think it's empty: https://github.com/maidsafe/routing/blob/fleming/src/chain/chain.rs#L614. Also it doesn't matter it's outdated. It doesn't need to contain the latest keys. Any keys whose prefix is compatible with the target prefix will do.

Reply:
```
        let trimmed_info = GenesisPfxInfo {
            first_info: self.gen_pfx_info.first_info.clone(),
            first_bls_keys: self.gen_pfx_info.first_bls_keys.clone(),
            first_state_serialized: Default::default(),
            first_ages: self.gen_pfx_info.first_ages.clone(),
            latest_info: self.chain.our_info().clone(),
            parsec_version: self.gen_pfx_info.parsec_version,
        };

        let src = Location::PrefixSection(*trimmed_info.first_info.prefix());
        let content = MessageContent::NodeApproval(trimmed_info);
```

We trimm it out for NodeApproval, I guess we missed doing that for GenesisUpdate, so new node will not have the info, but ones that have seen an elder change will. Not ideal.

Reply:
Would it make things easier if we did the work of making relocate like GenesisUpdate i.e don't have to sign it before it goes through Parsec? then we can just add whatever we need when we send the message.

Reply:
Also you are correct it does not matter that it is out of date since its history will always be kept by the destination even if we trim it eventually it should not matter for that case. (But Adults should really not have that knowledge to start with).

Reply:
> Would it make things easier if we did the work of making relocate like GenesisUpdate i.e don't have to sign it before it goes through Parsec? then we can just add whatever we need when we send the message.

Yes I think that's the plan. Not sure it makes sense to do it in this PR though.

Reply:
yes, it does not really make sense in this PR, could be if we wanted, but if it make that PR easier could be done in a PR that we rebase 2024 ontop?